### ISSUE TITLE
[onert] Performance issue in mobilenetv2 training

### ISSUE THREAD

**Issue by jyoungyun** [Samsung/ONE#12569]
### Performance issue

In my test environment,

Dataset: imagenet_a

epoch: 5

```
== training parameter ==
- learning_rate   = 0.001
- batch_size      = 10
- loss_info       = {loss = categorical crossentropy, reduction = sum over batch size}
- optimizer       = adam
========================
```

| step count<br>per 1 epoch<br>(data_length) | TensorFlow | onert_train(rel) |
|:--:|:--:|:--:|
| 1(10) | 4.9956 | 5.0758 |
| 10(100) | 13.8550 | 55.3730 |
| 100(1000) | 87.8630 | 539.7906 |

😮 

_If you need detailed information, please contact me. :)_

_original issue https://github.com/Samsung/ONE/issues/12325#issuecomment-1916498006_

**Comment by jyoungyun** [comment]
### 진행상황 2.27

- 용현님 구현한 내용을 확인해보자.

| step count<br>per 1 epoch<br>(data_length) | TensorFlow | onert_train(rel) |
|:--:|:--:|:--:|
| 1(10) | 5.0842 | 3.0051 |
| 10(100) | 12.4805 | 30.0293 |
| 100(1000) | 102.2641 | 303.4921 |

<details>
<summary>log</summary>

batch_size: 10
data_length: 10

tensorflow
```
Epoch 1/5
1/1 [==============================] - 4s 4s/step - loss: 7.2665 - accuracy: 0.0000e+00
Epoch 2/5
1/1 [==============================] - 0s 169ms/step - loss: 0.1066 - accuracy: 0.0000e+00
Epoch 3/5
1/1 [==============================] - 0s 171ms/step - loss: 0.0049 - accuracy: 0.0000e+00
Epoch 4/5
1/1 [==============================] - 0s 167ms/step - loss: 0.0030 - accuracy: 0.0000e+00
Epoch 5/5
1/1 [==============================] - 0s 169ms/step - loss: 0.0018 - accuracy: 0.0000e+00
==========================
Total time: 5.0842
```

onert_train(release)
```
./Product/out/bin/onert_train --modelfile ../Personal/models/mobilenet_v2/keras_model/mobilenetv2.circle --load_input:raw ../Personal/models/data/imagenet_a/test.input.10.bin --load_expected:raw ../Personal/models/data/imagenet_a/test.output.10.bin --epoch 5 --batch_size 10 --loss 2 --optimizer 2 --learning_rate 0.001 --loss_reduction_type 1
Model Expected Filename ../Personal/models/data/imagenet_a/test.output.10.bin
Model Input Filename ../Personal/models/data/imagenet_a/test.input.10.bin
Model Filename ../Personal/models/mobilenet_v2/keras_model/mobilenetv2.circle
== training parameter ==
- learning_rate   = 0.001
- batch_size      = 10
- loss_info       = {loss = categorical crossentropy, reduction = sum over batch size}
- optimizer       = adam
========================
Epoch 1/5 - time: 631.918ms/step - loss: [0] 8.5157
Epoch 2/5 - time: 614.231ms/step - loss: [0] 3.0827
Epoch 3/5 - time: 593.521ms/step - loss: [0] 0.5703
Epoch 4/5 - time: 560.376ms/step - loss: [0] 1.6380
Epoch 5/5 - time: 600.620ms/step - loss: [0] 1.5613
===================================
MODEL_LOAD   takes 13.4950 ms
PREPARE      takes 285.9840 ms
EXECUTE      takes 3005.1210 ms
- Epoch 1      takes 631.9180 ms
- Epoch 2      takes 614.2310 ms
- Epoch 3      takes 593.5210 ms
- Epoch 4      takes 560.3760 ms
- Epoch 5      takes 600.6200 ms
===================================
```

batch_size: 10
data_length: 100

tensorflow
```
Epoch 1/5
10/10 [==============================] - 6s 167ms/step - loss: 6.9907 - accuracy: 0.0000e+00
Epoch 2/5
10/10 [==============================] - 2s 166ms/step - loss: 1.1671 - accuracy: 0.0000e+00
Epoch 3/5
10/10 [==============================] - 2s 166ms/step - loss: 0.0885 - accuracy: 0.0000e+00
Epoch 4/5
10/10 [==============================] - 2s 166ms/step - loss: 0.0119 - accuracy: 0.0000e+00
Epoch 5/5
10/10 [==============================] - 2s 165ms/step - loss: 0.0057 - accuracy: 0.0000e+00
==========================
Total time: 12.4805
```

onert_train(release)
```
./Product/out/bin/onert_train --modelfile ../Personal/models/mobilenet_v2/keras_model/mobilenetv2.circle --load_input:raw ../Personal/models/data/imagenet_a/test.input.100.bin --load_expected:raw ../Personal/models/data/imagenet_a/test.output.100.bin --epoch 5 --batch_size
 10 --loss 2 --optimizer 2 --learning_rate 0.001 --loss_reduction_type 1
Model Expected Filename ../Personal/models/data/imagenet_a/test.output.100.bin
Model Input Filename ../Personal/models/data/imagenet_a/test.input.100.bin
Model Filename ../Personal/models/mobilenet_v2/keras_model/mobilenetv2.circle
== training parameter ==
- learning_rate   = 0.001
- batch_size      = 10
- loss_info       = {loss = categorical crossentropy, reduction = sum over batch size}
- optimizer       = adam
========================
Epoch 1/5 - time: 589.897ms/step - loss: [0] 7.2317
Epoch 2/5 - time: 610.394ms/step - loss: [0] 5.2931
Epoch 3/5 - time: 602.978ms/step - loss: [0] 4.6070
Epoch 4/5 - time: 597.920ms/step - loss: [0] 4.5122
Epoch 5/5 - time: 598.049ms/step - loss: [0] 4.2477
===================================
MODEL_LOAD   takes 5.2100 ms
PREPARE      takes 281.1990 ms
EXECUTE      takes 30029.3760 ms
- Epoch 1      takes 5898.9730 ms
- Epoch 2      takes 6103.9370 ms
- Epoch 3      takes 6029.7780 ms
- Epoch 4      takes 5979.2000 ms
- Epoch 5      takes 5980.4900 ms
===================================
```

batch_size: 10
data_length: 1000

tensorflow
```
Epoch 1/5
100/100 [==============================] - 23s 193ms/step - loss: 5.9272 - accuracy: 0.0000e+00
Epoch 2/5
100/100 [==============================] - 19s 193ms/step - loss: 5.0403 - accuracy: 0.0000e+00
Epoch 3/5
100/100 [==============================] - 19s 194ms/step - loss: 4.8337 - accuracy: 0.0000e+00
Epoch 4/5
100/100 [==============================] - 19s 194ms/step - loss: 4.6267 - accuracy: 0.0000e+00
Epoch 5/5
100/100 [==============================] - 21s 207ms/step - loss: 4.4541 - accuracy: 0.0000e+00
==========================
Total time: 102.2641
```

onert_train(release)
```
./Product/out/bin/onert_train --modelfile ../Personal/models/mobilenet_v2/keras_model/mobilenetv2.circle --load_input:raw ../Personal/models/data/imagenet_a/test.input.1000.bin --load_expected:raw ../Personal/models/data/imagenet_a/test.output.1000.bin --epoch 5 --batch_si
ze 10 --loss 2 --optimizer 2 --learning_rate 0.001 --loss_reduction_type 1
Model Expected Filename ../Personal/models/data/imagenet_a/test.output.1000.bin
Model Input Filename ../Personal/models/data/imagenet_a/test.input.1000.bin
Model Filename ../Personal/models/mobilenet_v2/keras_model/mobilenetv2.circle
== training parameter ==
- learning_rate   = 0.001
- batch_size      = 10
- loss_info       = {loss = categorical crossentropy, reduction = sum over batch size}
- optimizer       = adam
========================
Epoch 1/5 - time: 606.864ms/step - loss: [0] 5.9483
Epoch 2/5 - time: 601.767ms/step - loss: [0] 5.0921
Epoch 3/5 - time: 604.828ms/step - loss: [0] 5.0245
Epoch 4/5 - time: 606.615ms/step - loss: [0] 5.0058
Epoch 5/5 - time: 611.152ms/step - loss: [0] 5.0005
===================================
MODEL_LOAD   takes 9.4200 ms
PREPARE      takes 282.2300 ms
EXECUTE      takes 303492.1890 ms
- Epoch 1      takes 60686.4430 ms
- Epoch 2      takes 60176.6540 ms
- Epoch 3      takes 60482.8480 ms
- Epoch 4      takes 60661.5240 ms
- Epoch 5      takes 61115.2340 ms
===================================
```

**Comment by jyoungyun** [comment]
### 진행상황 2.27

- 헐, loss 가 틀어지고 있다..

**Comment by Aeren1564** [comment]
This task has been completed. Closing.
